cogs/login.py

```diff
@@ -0,0 +1,150 @@
+        selection_embed.description = selection["description"]
+        selection_embed.set_footer(text="Password will be auto deleted in 15 seconds")
+
+        async def return_callback(interaction: Any):
+            await previous.edit(embed=home, view=old_view)
+            await interaction.response.defer()
+
+        async def gen_callback(interaction: Any):
+            await interaction.response.send_message(
+                content=selection["password"], ephemeral=True, delete_after=15
+            )
+
+        async def email_callback(interaction: Any):
+            await interaction.response.send_message(
+                content=selection["email"], ephemeral=True, delete_after=15
+            )
+
+        return_button = Button(label="Return", emoji="⬅")
+        return_button.callback = return_callback
+
+        selection_link = Button(label="Go to Site", url=selection["link"])
+
+        gen_button = Button(
+            label="Generate Password", style=discord.ButtonStyle.green, emoji="🔐"  #type: ignore
+        )
+        gen_button.callback = gen_callback
+
+        email_button = Button(
+            label="Generate Email", style=discord.ButtonStyle.green, emoji="👤"     #type: ignore
+        )
+        email_button.callback = email_callback
+
+        view = View(timeout=60)
+
+        view.add_item(return_button)
+        view.add_item(selection_link)
+        view.add_item(email_button)
+        view.add_item(gen_button)
+
+        await previous.edit(embed=selection_embed, view=view)
+
+    @commands.command()
+    async def login(self, ctx: Context):
+        # Create Embed with Python, reads from json config file
+        config = Login.open_json()
+
+        if config is None:
+            embed = Embed()
+            embed.title = "JSON File Broken"
+            embed.description = "Please contact Bot Admins to Fix"
+            embed.color = Color.red()
+            await ctx.send(embed=embed)
+            return
+
+        embed = Embed()
+        embed.title = "Various Login Information"
+        embed.description = "Please do not share this information with anyone else!"
+        embed.set_footer(text="Email and password provided using selection menu")
+        embed.color = Color.blue()
+
+        select = Select(  # Select Method that Creates Selection Menu for Embed
+            placeholder="Choose a streaming service..."  # Placeholder Text
+        )
+
+        for i in config["login_info"]:
+            embed.add_field(
+                name=i["emoji_text"] + i["site"] + i["emoji_text"],
+                value="Provider: " + i["provider"],
+                inline=True,
+            )
+
+            select.add_option(
+                label=i["site"], emoji=None, description=i["site"] + " Login"
+            )
+
+        async def my_callback(interaction: Any):
+            if (
+                interaction.user == ctx.author
+            ):  # Interaction Author Must be Original Sender
```

Comment:
if you put these comments in a different line, the auto formatter wont do this weird placement thing.

it is doing it because it thinks the line is too long